swaglyrics_backend/issue_maker.py

```diff
@@ -413,46 +413,45 @@ def discord_instrumental_logger(song: str, artist: str,
-            return 'Issue already exists on the GitHub repo. \n' \
-                   'https://github.com/SwagLyrics/SwagLyrics-For-Spotify/issues'
-
-        # check if song, artist trivial (all letters and spaces)
-        if re.fullmatch(asrg, song) and re.fullmatch(asrg, artist):
-            return f'Lyrics for {song} by {artist} may not exist on Genius.\n' + gh_issue_text
-
-        # check if song exists on spotify and does not have lyrics on genius
-        if check_song(song, artist) and not check_stripper(song, artist):
-            with open('unsupported.txt', 'a', encoding='utf-8') as f:
-                f.write(f'{song} by {artist}\n')
-
-            issue = create_issue(song, artist, version, stripped)
-
-            if issue['status_code'] == 201:
-                logging.info(f'Created issue on the GitHub repo for {song} by {artist}.')
-                return 'Lyrics for that song may not exist on Genius. ' \
-                       f'Created issue on the GitHub repo for {song} by {artist} to investigate ' \
-                       f'further. \n{issue["link"]}'
-            else:
-                return f'Logged {song} by {artist} in the server.'
-
-        return "That's a fishy request, that song doesn't seem to exist on Spotify. \n" + gh_issue_text
+    song = request.form['song']
+    artist = request.form['artist']
+    stripped = stripper(song, artist)
+
+    try:
+        version = request.form['version']
+    except KeyError:
+        return update_text
+
+    logging.info(f"{song=}, {artist=}, {stripped=}, {version=}")
+    if version < '1.2.0':
+        return update_text
+
+    with open('unsupported.txt', 'r', encoding='utf-8') as f:
+        data = f.read()
+    if f'{song} by {artist}' in data:
+        return 'Issue already exists on the GitHub repo. \n' \
+               'https://github.com/SwagLyrics/SwagLyrics-For-Spotify/issues'
+
+    # check if song, artist trivial (all letters and spaces)
+    if re.fullmatch(asrg, song) and re.fullmatch(asrg, artist):
+        return f'Lyrics for {song} by {artist} may not exist on Genius.\n' + gh_issue_text
+
+    # check if song exists on spotify and does not have lyrics on genius
+    if check_song(song, artist) and not check_stripper(song, artist):
+        with open('unsupported.txt', 'a', encoding='utf-8') as f:
+            f.write(f'{song} by {artist}\n')
+
+        issue = create_issue(song, artist, version, stripped)
+
+        if issue['status_code'] == 201:
+            logging.info(f'Created issue on the GitHub repo for {song} by {artist}.')
+            return 'Lyrics for that song may not exist on Genius. ' \
+                   f'Created issue on the GitHub repo for {song} by {artist} to investigate ' \
+                   f'further. \n{issue["link"]}'
+        else:
+            return f'Logged {song} by {artist} in the server.'
+
+    return "That's a fishy request, that song doesn't seem to exist on Spotify. \n" + gh_issue_text
```

Comment:
I think this should be chopped into smaller functions for readability, like `check_version`, `check_song_exist` etc.

Reply:
It is already—there's check_stripper and check_song. Anyway, that is not the change here. I removed the redundant check if the request is POST or not because we already do that in the decorator

Reply:
It's up to you, but for me, function is too long